graphql-cli/src/test/resources/expectedGenCode/client.bal

```diff
@@ -6,20 +6,32 @@ public isolated client class GraphqlClient {
     public isolated function init(string serviceUrl, ConnectionConfig config) returns graphql:ClientError? {
         http:ClientConfiguration httpClientConfig = {
             httpVersion: config.httpVersion,
-            http1Settings: {...config.http1Settings},
-            http2Settings: config.http2Settings,
             timeout: config.timeout,
             forwarded: config.forwarded,
             poolConfig: config.poolConfig,
-            cache: config.cache,
             compression: config.compression,
             circuitBreaker: config.circuitBreaker,
-            retryConfig: config.retryConfig,
-            responseLimits: config.responseLimits,
-            secureSocket: config.secureSocket,
-            proxy: config.proxy,
-            validation: config.validation
+            retryConfig: config.retryConfig
         };
+        if config.http1Settings is ClientHttp1Settings {
+            ClientHttp1Settings settings = check config.http1Settings.ensureType(ClientHttp1Settings);
+            httpClientConfig.http1Settings = {...settings};
+        }
+        if config.http2Settings is http:ClientHttp2Settings {
+            httpClientConfig.http2Settings = check config.http2Settings.ensureType(http:ClientHttp2Settings);
+        }
+        if config.cache is http:CacheConfig {
+            httpClientConfig.cache = check config.cache.ensureType(http:CacheConfig);
+        }
+        if config.responseLimits is http:ResponseLimitConfigs {
+            httpClientConfig.responseLimits = check config.responseLimits.ensureType(http:ResponseLimitConfigs);
+        }
+        if config.secureSocket is http:ClientSecureSocket {
+            httpClientConfig.secureSocket = check config.secureSocket.ensureType(http:ClientSecureSocket);
+        }
+        if config.proxy is http:ProxyConfig {
+            httpClientConfig.proxy = check config.proxy.ensureType(http:ProxyConfig);
+        }
```

Comment:
Let's handle the error and return as a `ballerina/graphql` related error

Reply:
Fixed by adding `graphql:HttpError()` in https://github.com/ballerina-platform/graphql-tools/commit/820d69eccf3d162405d31e4797434fb404c1da4e